app/models/post.rb

```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :author, foreign_key: :author_id, class_name: 'User', counter_cache: :posts_counter
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def update_user_post_counter
+    author.increment!(:postscounter)
+  end
```

Comment:
- [ ] [OPTIONAL] You have done a great job with this method, but we can make it even cooler! To do this, you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the #update_user_post_counter method automatically every time a new post is been saved to the database 😎 Then make the update_user_post_counter method private.

**See below 👇**

```ruby
after_save :update_user_post_counter

private

def update_user_post_counter
  author.increment!(:postscounter)
end
```

Reply:
Thank you @codepantha
I will explore that.